File: frontend/src/types/ingest.ts
/**
 * 上海宇羲伏天智能科技有限公司出品
 *
 * 文件级注释：文档导入相关类型定义
 */

export interface IngestResponse {
  document_id: number;
  status: string;
  chunk_count: number;
}

export interface DBIngestRequest {
  connection_uri: string;
  table_name: string;
  content_column: string;
  metadata_columns?: string[];
}

export interface URLIngestRequest {
  url: string;
  tags?: string[];
}

/**
 * 函数级注释：任务状态类型
 * 内部逻辑：定义任务状态的联合类型
 */
export type TaskStatus = 'pending' | 'processing' | 'completed' | 'failed';

/**
 * 函数级注释：任务状态常量对象
 * 内部逻辑：提供任务状态的常量引用，便于代码中使用
 */
export const TaskStatus = {
  PENDING: 'pending' as const,
  PROCESSING: 'processing' as const,
  COMPLETED: 'completed' as const,
  FAILED: 'failed' as const
} as const;

export interface TaskResponse {
  id: number;
  file_name: string;
  status: TaskStatus;
  progress: number;
  error_message?: string;
  document_id?: number;
  created_at: string;
  updated_at: string;
}

export interface TaskListResponse {
  items: TaskResponse[];
  total: number;
}

File: code/app/schemas/ingest.py
"""
上海宇羲伏天智能科技有限公司出品

文件级注释：知识摄入相关的 Pydantic 模型
"""

from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from enum import Enum

class TaskStatusEnum(str, Enum):
    """
    类级注释：任务状态枚举
    属性：
        PENDING: 等待处理
        PROCESSING: 处理中
        COMPLETED: 已完成
        FAILED: 失败
    """
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"

class IngestResponse(BaseModel):
    """
    类级注释：摄入接口响应模型
    属性：
        document_id: 文档唯一标识
        status: 处理状态 (processing/completed/failed)
        chunk_count: 切分出的片段数量
    """
    document_id: int
    status: str
    chunk_count: int

class DBIngestRequest(BaseModel):
    """
    类级注释：数据库摄入请求模型
    属性：
        connection_uri: 数据库连接字符串 (如 sqlite:///test.db)
        table_name: 要同步的表名
        content_column: 包含知识内容的列名
        metadata_columns: 可选的其他列名列表，作为元数据存入
    """
    connection_uri: str
    table_name: str
    content_column: str
    metadata_columns: Optional[List[str]] = None

class URLIngestRequest(BaseModel):
    """
    类级注释：网页摄入请求模型
    属性：
        url: 目标网页的完整 URL 地址
        tags: 可选的元数据标签
    """
    url: str
    tags: Optional[List[str]] = None

class TaskResponse(BaseModel):
    """
    类级注释：任务响应模型
    属性：
        id: 任务ID
        file_name: 文件名
        status: 任务状态
        progress: 处理进度 (0-100)
        error_message: 错误信息
        document_id: 关联文档ID
        created_at: 创建时间
        updated_at: 更新时间
    """
    id: int
    file_name: str
    status: TaskStatusEnum
    progress: int
    error_message: Optional[str] = None
    document_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

class TaskListResponse(BaseModel):
    """
    类级注释：任务列表响应模型
    属性：
        items: 任务列表
        total: 总数
    """
    items: List[TaskResponse]
    total: int

